### ISSUE TITLE
Benchmark requests

### ISSUE THREAD

**Issue by ImmanuelSamuel** [ballercat/walt#112]
I was pleasantly surprised when I found this repo. However, please provide some benchmarks that can push people like me to start using this. Great work btw!

**Comment by ballercat** [comment]
Thank you for the suggestion.

I would like some benchmarks of course but first I would need some useful JS package which could be converted. It's a bit tough for me to implement due to time constraints. Eventually, this compiler would be self-hosted so there would be a good A/B comparison to be made there, on a large codebase.

There are some toy examples in the walt-explorer GitHub page. Like the naive Fibonacci example it is much faster than a JS equivalent. The JS equivalent might even hang indefinitely. 

I'll leave the issue open, maybe someone will come along and implement a few. If you have some ideas of benchmarks you'd like to see that would be helpful.

**Comment by maudnals** [comment]
Hi ballercat,
I'm new to walt but would love to help. My background is mostly frontend JS (and long ago C++ with OpenGL). 
Am I getting this right, that the goal of this issue is to provide convincing benchmarks of walt vs JS? Which is basically showcasing wasm's performance kick, with specific walt examples?
If so:
How about benchmarks around applying filters on images, would that make sense?
i.e. implementing https://www.html5rocks.com/en/tutorials/canvas/imagefilters/ in walt, and comparing performance on large images?

**Comment by ballercat** [comment]
Sounds good to me!

**Comment by maudnals** [comment]
Hi @ballercat ,   

Here are some note about [2], sorry, time flew :)  
Will look into [1] later on. 

Here's my experience when getting started with Walt. Context: I know JS. I'm new to walt, but au fait with Wasm (with Rust).  

### TL;DR:
- Onboarding part 1  = get the start project running): fantastic 🔥  
- Onboarding part 2 = actually writing my own walt code: things get harder. 
=> Walt's strength to me is its similarity to JS, so it could be leveraged more!

### Long version  

#### Onboarding part 1: scaffolding a walt project for the first time   

* [+] https://github.com/ballercat/walt/wiki/Walt-In-5-Minutes is dead simple si very nice. And it works. Low bareer to entry, the start project runs smoothly, and it feels just like a normal frontend project.
* [-] but: https://github.com/ballercat/walt/wiki/Walt-In-5-Minutes is a hybrid way to get started: there are some manual tasks to perform, but they don't help me understand better how a project is built, since they are common frontend workflow tasks. 
As a walt newcomer, I'd have preferred: 
  * Either a readymade boilerplate that I just need to npm install then run - I can look at the project structure myself afterwards. Not sure whether it's intentional not to have one, because walt is evolving, and maintaining it might be too much of a hassle at this stage?  
  * Or a more thorough version of "Walt in 5 minutes" (... a "Walt in 10 minutes" ;) ), that would detail a little further how walt works / how the walt file will be run.
  * Or both, offered as separate options. 
As a side note:  I can't imagine a specific case for JS newcomers to use walt, since they might as well use their own toolchains (C++/Rust), but maybe there's one! So: if we think of devs who are new to JS, they might be better off with a boilerplate as well, to first abstract them totally from webpack. 

#### Onboarding part 2: Writing walt code
* [-] No syntax highlighting for .walt in my editor VS Code (also looked the market extensions, no luck): obviously this is just a matter of time, but getting a suggestion there would definitely help!  
* [+] Walt _really_ looks like JS, that's great 
* [-] Because Walt really looks like JS, I'm tempted to write code such as: 
```javascript 
const arr: i32[] = [1, 2]; // this crashes
```
So I head to the Syntax wiki page; it's very detailed, but it's missing more examples.  
Suggestion: because the strength of Walt is exactly that it feels like JS, why not have a "snippets" doc where Walt code and its JS equivalent are displayed side-by-side?  
* [-] I'm still not clear about walt-JS interop. Is it also using wasm-bindgen? Maybe this relates to https://github.com/ballercat/walt/issues/114.    


What do you think?
